Callum's Branch

The aim of this Initial Data exploration is to investigate patterns that may exist within the dataset, which may be worth of further statistical investigation.

The below code imports all relevant libraries, and handles the initial loading of the dataset

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import datetime as dt

#Import the Data
dataset = pd.read_csv("../../datasets/Participant-03.csv")

#rename Columns names to uniform standard
dataset.rename(columns = {"Steps (count)": "Steps"}, inplace = True)

#Drop any rows with null Values, and print the head and tail of the dataset for intial investigation
dataset.dropna()
print(dataset.head)



<bound method NDFrame.head of                  Start            Finish        Steps
0     31/12/2021 23:00   01/01/2022 0:00     0.000000
1      01/01/2022 0:00   01/01/2022 1:00    14.000000
2      01/01/2022 1:00   01/01/2022 2:00     0.000000
3      01/01/2022 2:00   01/01/2022 3:00     0.000000
4      01/01/2022 3:00   01/01/2022 4:00     0.000000
...                ...               ...          ...
6015  08/09/2022 13:00  08/09/2022 14:00     0.000000
6016  08/09/2022 14:00  08/09/2022 15:00   893.959035
6017  08/09/2022 15:00  08/09/2022 16:00   534.040965
6018  08/09/2022 16:00  08/09/2022 17:00   390.020268
6019  08/09/2022 17:00  08/09/2022 18:00  1191.979732

[6020 rows x 3 columns]>


The above results show the shape of the data (6020 records), as well as the makeup of the inputed columns, especially the Start and End columns, which will be reformatted in the next section. 

The goal of the next section is to investigate whether there are board trends based on days of the week in terms of steps taken.

In [3]:
#Transform String into DataTime format for Start and Finish Columns
dataset['Start'] = pd.to_datetime( dataset['Start'], infer_datetime_format = True)
dataset['Finish'] = pd.to_datetime( dataset['Finish'], infer_datetime_format = True)

#Create a new column with the starting day of each sample 1 hours period
dataset['Day'] = dataset['Start'].dt.day_of_week
#dataset.info()

weekdays = dataset[(dataset['Day'] != 6) & (dataset['Day'] != 7)]
weekends = dataset[(dataset['Day'] == 6) | (dataset['Day'] == 7)]

weekdays['Steps'].describe()



count    5155.000000
mean      252.261106
std       548.980608
min         0.000000
25%         0.000000
50%         0.000000
75%       148.000000
max      4828.967310
Name: Steps, dtype: float64

In [4]:
weekends['Steps'].describe()
weekends.head()

Start              Finish  Steps  Day
25 2022-01-02 00:00:00 2022-02-01 01:00:00    0.0    6
26 2022-01-02 01:00:00 2022-02-01 02:00:00   16.0    6
27 2022-01-02 02:00:00 2022-02-01 03:00:00    0.0    6
28 2022-01-02 03:00:00 2022-02-01 04:00:00    0.0    6
29 2022-01-02 04:00:00 2022-02-01 05:00:00    0.0    6

These results show that there is ia significant difference across the different days, with a clear dip during Saturday and Sunday, with a mid-week "hump" in the number of steps walked. This lines up with previous literature on the subject. This initial Analysis informs the further questions that can be tested on this dataset, including why there is such a large weekend reduction, and why wednesday is the most active. Whether these trends hold across seasons may be another question for our research.